virttest/utils_time.py

```diff
@@ -0,0 +1,243 @@
+        tz_list = tz_list.splitlines()
+        for line in tz_list:
+            if not line:
+                continue
+            og = re.search(r'(?:\(UTC([+|-]\d{2}:\d{2})?)', line,
+                           re.M | re.I)
+            if og and og.group(1):
+                og_ = re.sub(r'(\d{2}):(\d{2})', r'\1\2', og.group(1))
+                guest_tz.append(og_)
+                continue
+            elif og:
+                guest_tz.append("+0000")
+                continue
+            og_value = re.search(r'(\S+(?:\s\S+)*)', line, re.M | re.I)
+            if og_value:
+                guest_tz.append(og_value.group(0))
+            guest_tzs.append(guest_tz)
+            guest_tz = []
+
+        for value in guest_tzs:
+            if o_guest == value[1] and value[0] == zonecode:
+                return (True, session)
+
+        # Windows guest's timezone not match host's timezone
+        # get the timezone value for windows guest
+        tz_value = ""
+        for value in guest_tzs:
+            if zonecode == value[0]:
+                tz_value = value[1]
+                break
+        status = session.cmd_status(set_tz_cmd % tz_value)
+
+    if not status:
+        session = vm.reboot(session=session, timeout=600)
+        return (True, session)
+    else:
+        logging.warn("Fail to update timezone in guest")
+        return (False, session)
+
+
+def set_guest_clocksource(session, list_guest, expected):
+    """
+    Set clocksource for linux guest.
+
+    :param session: VM session.
+    :param list_guest: Avaialble clocksource list for guest.
+    :param expected: clocksource you expected.
+
+    :return: True if set clocksource successfully.
+    """
+    clocksource_dir = "/sys/devices/system/clocksource/clocksource0"
+    clocksource_dir += "current_clocksource"
+    try:
+        if list_guest and expected in list_guest:
+            if session.cmd_status("echo %s > %s" % (expected,
+                                  clocksource_dir)):
+                return True
+        else:
+            logging.warn("Expected clocksource not supported by guest.")
+            return False
+    except Exception:
+        return False
+
+
+def get_guest_clocksource(session, list_guest):
+    """
+    Get current clocksource in use.
+
+    :param session: VM session.
+    :param list_guest: Avaialble clocksource list of guest.
+    :param source: Clocksource want to set guest to.
+
+    :return: clocksource get.
+    """
+    clocksource_dir = "/sys/devices/system/clocksource/clocksource0/"
+    clocksource_dir += "current_clocksource"
+    try:
+        cs = session.cmd_output("cat %s" % clocksource_dir)
+        if cs in list_guest:
+            pass
+        else:
+            logging.warn("Available clocksource list may outdate.")
+        return cs
+    except Exception:
+        return None
+
+
+def get_guest_availble_clocksource(session, clocksource_list, dy_avail=False):
```

Comment:
`get_guest_availble_clocksource` and `get_host_availble_clocksource` can be clubbed to `get_available_clock_source()`